### PR TITLE
lncli: allow configurable timeout in send payment

### LINKED ISSUE #4686: Feature Request: Allow paymentTimeoutSeconds to be configurable in lncli
### Background

Right now `paymentTimeoutSeconds` is hardcoded in the cli to 60 seconds for `sendpayment`. I am requesting that it's configurable as I would like to increase this value, unless there's a non obvious reason why it's limited. 

https://github.com/lightningnetwork/lnd/blob/1bd211a7228d2f13ecaf514f611738837c057d94/cmd/lncli/cmd_pay.go#L31

https://github.com/lightningnetwork/lnd/blob/1bd211a7228d2f13ecaf514f611738837c057d94/cmd/lncli/cmd_pay.go#L339

### PR DESCRIPTION
**Pull request by carlaKC**
Fixes #4686

### CODE REVIEW

File: cmd/lncli/cmd_pay.go [hunk 1]
```diff
@@ -336,7 +343,13 @@ func sendPaymentRequest(ctx *cli.Context,
 	}
 
 	req.CltvLimit = int32(ctx.Int(cltvLimitFlag.Name))
-	req.TimeoutSeconds = paymentTimeoutSeconds
+
+	pmtTimeout := ctx.Duration("timeout")
+	if pmtTimeout == 0 {
```
Comment (halseth) [inline cmd/lncli/cmd_pay.go:348]: should check not negative?

**Review by halseth — COMMENTED** [review]
Useful! As far as I can remember, there is no specific reason why we haven't exposed this on the CLI before.

File: cmd/lncli/cmd_pay.go [hunk 2]
```diff
@@ -336,7 +343,13 @@ func sendPaymentRequest(ctx *cli.Context,
...
+		return errors.New("payment timeout must be non-zero")
+	}
+
+	req.TimeoutSeconds = int32(pmtTimeout.Seconds())
```
Comment (bhandras) [inline cmd/lncli/cmd_pay.go:352]: nit: extra new line after this line

File: cmd/lncli/cmd_pay.go [hunk 3]
```diff
@@ -336,7 +343,13 @@ func sendPaymentRequest(ctx *cli.Context,
 	}
 
 	req.CltvLimit = int32(ctx.Int(cltvLimitFlag.Name))
-	req.TimeoutSeconds = paymentTimeoutSeconds
+
+	pmtTimeout := ctx.Duration("timeout")
```
Comment (bhandras) [inline cmd/lncli/cmd_pay.go:347]: nit++: why not call this `paymentTimeout`?

File: cmd/lncli/cmd_pay.go (hunk 2 above)
Comment (carlaKC) [reply cmd/lncli/cmd_pay.go:351]: preexisting, there are newlines between `AllowSelfPayment` and `MaxParts` below as well.

File: cmd/lncli/cmd_pay.go (hunk 3 above)
Comment (carlaKC) [reply cmd/lncli/cmd_pay.go:347]: done 👍

File: cmd/lncli/cmd_pay.go (hunk 1 above)
Comment (carlaKC) [reply cmd/lncli/cmd_pay.go]: indeed, didn't think that `-1s` was possible input for a duration flag, but it is, check added

**Review by bhandras — APPROVED** [review]
LGTM :+1:

**Review by halseth — APPROVED** [review]
LGTM 👍

**Pull request merged**